### PR TITLE
Completed sgt-back-end. | Rakesh

### PR DESCRIPTION
**Pull request by bhaisingh**
image
image
image
image
image

### CODE REVIEW

File: sgt-back-end/index.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        });
+    });
+});
+
+app.use(express.json());
+
+//nserts a new grade into the "grades" table and returns the created grade.
+app.post('/api/grades', (req, res, next) => {
+    if (req.body.name === undefined || req.body.course === undefined || req.body.score === undefined || 
+        typeof parseInt(req.body.score) !== 'number' || req.body.score % 1 !== 0 || req.body.score < 1 || req.body.score > 100 ) {
```
[the remark this answers is not in the document]
Comment (TimelikeClosure) [reply sgt-back-end/index.js:54]: However, `0` _should_ be a valid `score`:

---

image

File: sgt-back-end/index.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            "score" = $3
+        where "gradeId" = $4
+        returning *
+     `;
+    const values = [req.body.name, req.body.course, req.body.score, req.params.gradeId]
+
+    db.query(sql, values)
+        .then(result => {
+            if (result.rows.length > 0) {
+                res.status(201).json(result.rows)
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:111]: A `201 Created` response should only be sent when a new record is created, not when an existing one is updated.

File: sgt-back-end/index.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        delete from grades
+        where "gradeId" = $1
+        returning *
+     `;
+    const values = [req.params.gradeId]
+
+    db.query(sql, values)
+        .then(result => {
+            if (result.rows.length > 0) {
+                res.status(204).json(result.rows)
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:146]: A `204 No Content` response does not (and should not) include a response body.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (bhaisingh) [reply sgt-back-end/index.js]: change logic to consider 0 as valid score

File: sgt-back-end/index.js (hunk 2 above)
Comment (bhaisingh) [reply sgt-back-end/index.js]: Change return code to 200 when record is updated

File: sgt-back-end/index.js (hunk 3 above)
Comment (bhaisingh) [reply sgt-back-end/index.js]: Change logic to not return the deleted row

**Review by TimelikeClosure — APPROVED** [review]

**Comment by bhaisingh** [comment]
image
image
image
image
image

**Pull request merged**